scripts/revng-check-conventions

```diff
@@ -100,134 +109,244 @@ while [[ $# > 0 ]]; do
     esac
 done
 
+# Retrive files by extension or shebang
+# $1: extension to match
+# $2: shebang to match
+function find_by_ext_or_shebang() {
+    readarray -t ext_files < <($GIT_LS_FILES | grep "\.${1}$")
```

Comment:
* Please make all variable UPPERCASE.
* I'd honestly be happier to have a check on all `git ls-files` for the absence of space/newline and then just go the old way. I understand you want to go The Right Way™, but it's mostly OCD, if we have files with spaces or anything else, it's just a bug.

Reply:
For issue 1: All variables are uppercase now
For issue 2:
I've added a whitespace check before any file operations
I've tried converting everything to newline but a lot of things look uglier, with a lot `if [[ $(wc -l <<< "$FILES") -eq 0 ]]` and `tr '\n' ' '` that also spawn an extra process
also `${<var>[@]}` guarantees that each element is quoted before being passed as an argument, which is better
I've tried to get to a compromise by declaring all the `_FILES` variables together (to remove possible redundancies) and to have `ALL_FILES_N` and `FILES_N` (N is for newline) to allow easier `grep`-ing if needed

Moreover not quoting variables to explicitly allow splitting is [considered bad practice](https://github.com/koalaman/shellcheck/wiki/SC2086) by shellcheck, which would mean putting an ignore each time we invoke a function